src/index.js

```diff
@@ -0,0 +1,104 @@
+var sha1 = require('crypto').createHash('sha1');
+
+var extend = require('./utils/extend');
+var elementToString = require('./render');
+
+/**
+ * @typedef {Object} RenderElement
+ * @property {Function|String} element.component Renderer function or name of tag.
+ * @property {Object} element.props Properties list.
+ * @property {String[]|String[][]|Number[]|Number[][]} element.children Children.
+ */
+
+module.exports = {
+    /**
+     * @param {Object} decl React component declaration.
+     * @returns {Function} render
+     */
+    createClass: function (decl) {
+        var executedDecl = {
+            props: decl.getDefaultProps ? decl.getDefaultProps() : {},
+            state: decl.getInitialState ? decl.getInitialState() : {},
+
+            setState: function (data) {
+                this.state = extend(this.state, data);
+            }
+        };
+
+        var mixins = Array.isArray(decl.mixins) ? decl.mixins : [];
+        var instanceBlank = extend.apply(this, mixins.concat([decl, executedDecl]));
+
+        var hasComponentCache = typeof decl.getCacheKey === 'function';
+        var componentCachePrefix = hasComponentCache ? sha1.update(JSON.stringify(decl)).digest('hex') : '';
```

Comment:
In most cases `JSON.stringify(decl)` will be equal `{}`
not everyone uses `displayName`